app/main/posts/modify/post-editor.directive.js

```diff
@@ -118,6 +121,14 @@ function PostEditorController(
 
             // Initialize values on post (helps avoid madness in the template)
             attributes.map(function (attr) {
+                // Create associated media entity
+                if (attr.input === 'upload') {
+                    var media = {};
+                    if ($scope.post.values[attr.key]) {
+                        media = $scope.post.values[attr.key][0];
+                    }
+                    $scope.medias[attr.key] = media;
```

Comment:
What is in `media` at this point? just an ID? or an object?

This seem problematic because we're not keeping 2 copies of the media items around.. could we achieve the same thing just by grabbing a list of all the 'upload' type attributes?